scripts/spatial-algorithms/cdda174/script.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+MASON_NAME=spatial-algorithms
+MASON_VERSION=cdda174
+MASON_HEADER_ONLY=true
+
+. ${MASON_DIR}/mason.sh
+
+function mason_load_source {
+    mason_download \
+        https://github.com/mapbox/${MASON_NAME}/tarball/${MASON_VERSION} \
+        1f0c5a41c42a33295dc251ef540ce6f6c1e8d5da
+
+    mason_extract_tar_gz
+
+    export MASON_BUILD_PATH=${MASON_ROOT}/.build/mapbox-${MASON_NAME}-${MASON_VERSION}
+}
+
+function mason_compile {
+    mkdir -p ${MASON_PREFIX}/include/
+    cp -r include/mapbox ${MASON_PREFIX}/include/mapbox
+    cp -r include/boost ${MASON_PREFIX}/include/boost
```

Comment:
Unrelated to this PR/package version, but want to highlight that this `cp -r include/boost ${MASON_PREFIX}/include/boost` is a bit freaky. I added this in the first PR for the mason package and it relates to how spatial-algorithms vendors some boost geometry related "extension" headers that @artemp uses in spatial-algorithms but which don't (yet) get distributed with the normal boost install. Highlighting this because it is unclear how this will play with mason linking into `./mason_packages/.link/include/boost` when both the official boost package and spatial-algorithms are installed. Will one win out over the other, or will their unique paths both be linked by file. /cc @millzpaugh because this will be an important stress test for mason-js linking: to ensure this works.